src/hardware/openbench-logic-sniffer/protocol.c

```diff
@@ -20,6 +20,13 @@
 #include <config.h>
 #include "protocol.h"
 
+#define RETURN_ON_ERROR(statement) \
+	do{ \
+		int retval; \
+		if ((retval=(statement)) != SR_OK)\
+			return retval; \
+	} while(0)
+
```

Comment:
Yeah, saves some boilerplate. I wonder if it wasn't better to add this macro to 'libsigrok-internal.h', so others drivers can benefit, too?

Reply:
Makes sense to me! I introduce it to that file in the new commit 2173e9c7a08f6733a0320a9d22632611379f592c. If there's a better place to put it inside the file, please tell me.